File: app/src/main/java/com/mirash/familiar/model/CredentialsItem.java
package com.mirash.familiar.model;

import androidx.annotation.NonNull;

import com.mirash.familiar.db.Credentials;

/**
 * @author Mirash
 */
public class CredentialsItem implements ICredentials {
    private final Credentials credentials;
    private boolean isPasswordVisible;

    public CredentialsItem(Credentials credentials) {
        this.credentials = credentials;
    }

    @Override
    public String getTitle() {
        return credentials.getTitle();
    }

    @Override
    public String getLink() {
        return credentials.getLink();
    }

    @Override
    public String getLogin() {
        return credentials.getLogin();
    }

    @Override
    public String getPassword() {
        return credentials.getPassword();
    }

    @Override
    public String getPin() {
        return credentials.getPin();
    }

    @Override
    public String getDetails() {
        return credentials.getDetails();
    }

    @Override
    public int getId() {
        return credentials.getId();
    }

    @Override
    public int getPosition() {
        return credentials.getPosition();
    }

    public void setPasswordVisible(boolean passwordVisible) {
        isPasswordVisible = passwordVisible;
    }

    public boolean isPasswordVisible() {
        return isPasswordVisible;
    }

    @NonNull
    @Override
    public String toString() {
        return credentials.toString();
    }

    @Override
    public boolean isAlike(String query) {
        return credentials.isAlike(query);
    }
}

File: app/src/main/java/com/mirash/familiar/model/credentials/CredentialsItem.kt
package com.mirash.familiar.model.credentials

import kotlinx.serialization.Serializable

/**
 * @author Mirash
 */
@Serializable
class CredentialsItem(private val credentials: ICredentials) : ICredentials {
    var isPasswordVisible = false

    override val title: String?
        get() = credentials.title
    override val link: String?
        get() = credentials.link
    override val login: String?
        get() = credentials.login
    override val email: String?
        get() = credentials.email
    override val phone: String?
        get() = credentials.phone
    override val password: String?
        get() = credentials.password
    override val pin: String?
        get() = credentials.pin
    override val details: String?
        get() = credentials.details
    override val id: Long
        get() = credentials.id
    override val position: Int
        get() = credentials.position

    override fun toString(): String {
        return credentials.toString()
    }

    override fun isAlike(query: String): Boolean {
        return credentials.isAlike(query)
    }
}
